packages/api/utils/shapeValidator.ts

```diff
@@ -18,7 +18,10 @@ export function validateRequestShape<T>(
   event: ApiGatewayEventParsed<T>,
   ...extraValidators: Array<(obj: unknown) => obj is T>
 ): SetupSuccess<T> {
-  if (!Object.values(event.pathParameters as APIGatewayProxyEventPathParameters).every(isValidUuidV4)) {
+  if (
+    event.pathParameters &&
+    !Object.values(event.pathParameters as APIGatewayProxyEventPathParameters).every(isValidUuidV4)
+  ) {
```

Comment:
Why change this?
Have you tested if this breaks the createApplication and createEnvironment requests?

Reply:
```
if(event.pathParameters)
```
Doesn't need to be included because
```
    x-amazon-apigateway-request-validator: "request-params-validator"
```
Catches empty path parameters on the API gateway level.

Reply:
This was changed because requests to this operation shouldn't have any path parameters.  I don't want to catch empty path parameters, I want to allow empty path parameters.  Request body shape checking was failing validation when path parameters were not included.

Reply:
This should not alter the behavior of `createApplication` or `createEnvironment` because the presence of the path parameters is merely checked prior to their individual evaluation in the remainder of the boolean logic.

The initial presence check allows my operation to not provide path parameters, but continues to allow their evaluation by other operations.

What's more, the unit tests in `shapeValidator.test.ts` cover the altered function.

Reply:
I tried it out on my deployed stack, which has the `createApplication` and `createEnvironment`, and it worked as expected.